src/utils/functions.ts

```diff
@@ -0,0 +1,39 @@
+import * as fs from "fs";
+import path from "path";
+import { window, workspace } from "vscode";
+
+export const getSourceName = (contractTitle: string) => {
+  if (workspace.workspaceFolders === undefined) {
+    window.showErrorMessage("please open solidity project to work");
+    return;
+  }
+
+  const path_ = workspace.workspaceFolders[0].uri.fsPath;
+  if (fs.existsSync(path.join(path_, "cache", "solidity-files-cache.json"))) {
+    const file = JSON.parse(
+      JSON.stringify(
+        fs.readFileSync(
+          path.join(path_, "cache", "solidity-files-cache.json"),
+          { encoding: "utf-8" }
+        )
+      )
+    );
+    const data = JSON.parse(file);
+    const filePathArray: any = Object.values(data.files).filter(
+      (data: any) => data.artifacts[0] === contractTitle
+    );
+
+    const sourceName: string = filePathArray[0].sourceName;
+
+    return fs.existsSync(
+      path.join(
+        path_,
+        "artifacts",
+        sourceName,
+        `${contractTitle}_deployed_address.json`
```

Comment:
We can read contract addressed from the `ethcode API ` we dont need to use  node.js `fs` module . If there are any more requirements on the API we can develop that .

Reply:
Here we are checking `fileName_deployed_address.json` file for the contract that we are selecting is present or not.

Reply:
we have an api for reading the contract address probably we can add use that if no data is retuned that means no address is available and if the file is not present the contract would not show  in the list contract api

Reply:
![image](https://user-images.githubusercontent.com/72294477/228842109-7fdf3d31-3b22-420f-9e3c-8f68c94a73eb.png)